### ISSUE TITLE
[Feature]Any plan to add Dynamic Graph support in TF?

### ISSUE THREAD

**Issue by yangjunpro** [tensorflow/tensorflow#12321]
We know that TF follows the define-and-run paradigm, while some other DL frameworks support define-by-run paradigm, such as pyTorch and Chainer. 

"define-and-run" actually reduce the mind-set overhead for DL modeling guys due to that it can help reduce the try-and-feedback loop.

MxNet(another originally define-and-run DL framework) already adds support for dynamic graph through Gluon(https://github.com/zackchase/mxnet-the-straight-dope).

As far as I know, there are TensorFlow Fold project(https://github.com/tensorflow/fold) which provides dynamic batching support, but it doesn't provides the define-by-run support.

Is there any plan for TFer to add the dynamic graph support in the near future?
If not, my team is going to work on this stuff, before jumping in, I just want to make sure our energies  are invested on unique features rather than duplicated ones. 

Thanks

**Comment by carlthome** [comment]
Try `import tensorflow.contrib.imperative as tf` for something that works somewhat more like "define-by-run". https://github.com/tensorflow/tensorflow/tree/master/tensorflow/contrib/imperative

Also, one might wonder what this is: https://github.com/tensorflow/tensorflow/tree/master/tensorflow/python/eager

**Comment by skye** [comment]
@asimshankar @alextp

**Comment by asimshankar** [comment]
Thanks for the note @yangjunpro . 

We hear this feature request, and are looking into providing a solution. There's nothing actionable yet, but we'll have more information on it as we develop it.  Thanks!

**Comment by yangjunpro** [comment]
@asimshankar May I know the rough release plan as to the support for the dynamic graph?

Thanks

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by asimshankar** [comment]
[Eager execution](https://research.googleblog.com/2017/10/eager-execution-imperative-define-by.html) is relevant to this thread.

Closing this issue. Please feel encouraged to try eager execution out and file more detailed feature requests / bug reports. Thanks!
